pkg/collector/pipeline.go

```diff
@@ -45,11 +46,27 @@ func (c nowClock) Now() time.Time {
 	return time.Now()
 }
 
+// ContextGetter extracts a context from a http.Request. This allows for more
+// complex logic for getting a context beyond r.Context()
+type ContextGetter interface {
+	Context(r *http.Request) context.Context
+}
+
+// DefaultContextGetter implements a ContextGetter that gets the context
+// contained directly within the request
+type DefaultContextGetter struct{}
```

Comment:
This can be package-local (`defaultContextGetter`)

Reply:
Done.